src/interpreter.ts

```diff
@@ -0,0 +1,104 @@
+
+//
+// The Interpreter watches a directory where scanned ballot images will appear
+// and process/interpret them into a cast-vote record.
+//
+
+import * as chokidar from 'chokidar'
+import * as ImageJS from 'image-js'
+import jsQR from "jsqr"
+
+import {BallotCallbackFunction, Ballot, BallotStyle, CandidateContest, Contest, Contests, Election} from './types'
+
+let watcher:chokidar.FSWatcher, election:Election
+
+export function init(e:Election, directoryToWatch:string, callback:BallotCallbackFunction) {
```

Comment:
What do you think about passing an object as a single param if there are more than 2 params?

Reply:
Feels like an extra type just for the sake of ... naming the arguments? I guess, doesn't feel like a clear win.

Reply:
Ok, just note that without this it requires readers of the code to remember what each param is when this function is called later. How many params are too many?

Reply:
in the new version, it's now reduced to just two arguments: a `path`, and a `ballotCallback`. I think that's probably simpler and reasonably future-proof rather than add an interface, export it, etc?